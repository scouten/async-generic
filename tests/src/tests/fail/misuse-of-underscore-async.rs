use async_generic::async_generic;

#[async_generic]
fn do_stuff() -> String {
    if _async && true {
        // ERROR: _async must stand alone in expression
        unreachable!();
    } else {
        "not async".to_owned()
    }
}

async fn my_async_stuff() -> String {
    "async".to_owned()
}

fn main() {}
