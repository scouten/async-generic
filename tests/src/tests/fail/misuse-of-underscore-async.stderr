error[E0425]: cannot find value `_async` in this scope
 --> src/tests/fail/misuse-of-underscore-async.rs:5:8
  |
5 |     if _async && true {
  |        ^^^^^^ not found in this scope
